### PR TITLE
ON-32908 # Changed routes to be optional

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: typescript/api.ts [hunk 1]
```diff
@@ -60,10 +97,7 @@ export type APIDeploymentPayload = {
   variables: {
     [key: string]: string
   }
-  routes: Array<{
-    module: string
-    route: string
-  }>
+  routes: APIEnvironmentRoute[]
```
Comment (divporter) [inline typescript/api.ts:100]: should this be optional??

File: typescript/api.ts [hunk 2]
```diff
@@ -5,11 +5,54 @@ export type APIEnvironmentRoute = {
...
+    }
+    /** The schedule configuration in a structure format parsed from a cron expression */
+    schedule?: APIEnvironmentSchedule
+  }
+
 export type APIEnvironment = {
   apiId: string
   environment: string
   lastDeployment: string
   routes: APIEnvironmentRoute[]
```
Comment (divporter) [inline typescript/api.ts:54]: should this be optional?

File: typescript/api.ts (hunk 1 above)
Comment (mymattcarroll) [reply typescript/api.ts:100]: it is currently required and will require more changes the API to make it optional. No breaking changes and no work to update is prefereable to no breaking changes and work to update IMO.

File: typescript/api.ts (hunk 2 above)
Comment (mymattcarroll) [reply typescript/api.ts:54]: we can make it an empty array

**Pull request merged**